x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileDocument.java

```diff
@@ -110,6 +126,29 @@ static ProfileDocument fromSubject(Subject subject) {
         );
     }
 
+    static String computeBaseUidForSubject(Subject subject) {
+        final MessageDigest digest = MessageDigests.md5();
```

Comment:
The length of final UID with `md5` digest is 20. It is 32 with `sha1` and 47 with `sha256`. I think `md5` should be sufficient for this use case. It is also available in BCFIPS lib.

Reply:
I would use `sha256` for any new code, simply to avoid discussions like this one.